Subject: [Buildroot] svn commit: trunk/buildroot/package/openssl
Date: Tue Jul 8 15:34:03 UTC 2008 
From: John Voltz <john.voltz at gmail.com>

Here's a patch to add SSL certificates to buildroot. It builds them from
mozilla's svn, so the certs are not likely to have been tampered with. It's
really pretty slick.


Index: buildroot/package/ca-certificates/mkcabundle.pl
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/ca-certificates/mkcabundle.pl	2012-06-05 20:18:51.000000000 +0000
@@ -0,0 +1,41 @@
+#!/usr/bin/perl -w
+#
+# Used to regenerate ca-bundle.crt from the Mozilla certdata.txt.
+# Run as ./mkcabundle.pl > ca-bundle.crt
+#
+
+my $cvsroot = ':pserver:anonymous@cvs-mirror.mozilla.org:/cvsroot';
+my $certdata = 'mozilla/security/nss/lib/ckfw/builtins/certdata.txt';
+
+open(IN, "cvs -d $cvsroot co -p $certdata|")
+    || die "could not check out certdata.txt";
+
+my $incert = 0;
+
+print<<EOH;
+# This is a bundle of X.509 certificates of public Certificate
+# Authorities.  It was generated from the Mozilla root CA list.
+#
+# Source: $certdata
+#
+EOH
+
+while (<IN>) {
+    if (/^CKA_VALUE MULTILINE_OCTAL/) {
+        $incert = 1;
+        open(OUT, "|openssl x509 -text -inform DER -fingerprint")
+            || die "could not pipe to openssl x509";
+    } elsif (/^END/ && $incert) {
+        close(OUT);
+        $incert = 0;
+        print "\n\n";
+    } elsif ($incert) {
+        my @bs = split(/\\/);
+        foreach my $b (@bs) {
+            chomp $b;
+            printf(OUT "%c", oct($b)) unless $b eq '';
+        }
+    } elsif (/^CVS_ID.*Revision: ([^ ]*).*/) {
+        print "# Generated from certdata.txt RCS revision $1\n#\n";
+    }
+}
Index: buildroot/package/ca-certificates/ca-certificates.mk
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/ca-certificates/ca-certificates.mk	2012-06-05 21:02:50.000000000 +0000
@@ -0,0 +1,25 @@
+#############################################################
+#
+# ca-certificates
+#
+#############################################################
+
+package/ca-certificates/ca-bundle.crt:
+	package/ca-certificates/mkcabundle.pl > $@
+
+$(TARGET_DIR)/usr/lib/ssl/cert.pem: package/ca-certificates/ca-bundle.crt
+	mkdir -p $(TARGET_DIR)/usr/lib/ssl/certs/
+	mkdir -p $(TARGET_DIR)/usr/lib/ssl/CA/private
+	cp package/ca-certificates/ca-bundle.crt $(TARGET_DIR)/usr/lib/ssl/certs
+	ln -s certs/ca-bundle.crt $(TARGET_DIR)/usr/lib/ssl/cert.pem
+
+ca-certificates: $(TARGET_DIR)/usr/lib/ssl/cert.pem
+
+#############################################################
+#
+# Toplevel Makefile options
+#
+#############################################################
+ifeq ($(strip $(BR2_PACKAGE_CA_CERTIFICATES)),y)
+TARGETS+=ca-certificates
+endif
Index: buildroot/package/ca-certificates/Config.in
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ buildroot/package/ca-certificates/Config.in	2012-06-05 21:04:29.000000000 +0000
@@ -0,0 +1,5 @@
+config BR2_PACKAGE_CA_CERTIFICATES
+	bool "ssl certificates"
+	default n
+	help
+	    SSL certificates for OpenSSL
Index: buildroot/package/Config.in
===================================================================
--- buildroot.orig/package/Config.in	2012-05-30 21:23:07.000000000 +0000
+++ buildroot/package/Config.in	2012-06-05 21:05:35.000000000 +0000
@@ -390,6 +390,7 @@
 source "package/libcgi/Config.in"
 source "package/libcgicc/Config.in"
 source "package/libcurl/Config.in"
+source "package/ca-certificates/Config.in"
 source "package/libdnet/Config.in"
 source "package/libesmtp/Config.in"
 source "package/libeXosip2/Config.in"
